src/todo.js

```diff
@@ -0,0 +1,124 @@
+import Project from './project';
+import * as utils from './utility';
+
+
+export default class Todo {
+  static mainList = [];
+
+  static priorityText = ['Low', 'Medium', 'High'];
+
+  constructor(title, description, dueDate, priority, project) {
+    this.title = title;
+    this.description = description;
+    this.dueDate = dueDate;
+    this.priority = priority;
+    this.project = project;
+    this.index = Todo.mainList.length;
+    this.done = false;
+    this.addToProject(project);
+    Todo.mainList.push(this);
+  }
+
+  addToProject(id) {
+    Project.mainList[id].addTodo(this);
+  }
+
+  changeAttr(attr, string) {
+    this[attr] = string;
+  }
+
+  findProps() {
+    return Object.values(this);
+  }
+
+  formatProps() {
+    const props = this.findProps();
+    const arr = [];
+    for (let i = 0; i < 3; i += 1) {
+      arr.push(props[i]);
+    }
+    arr.push(Todo.priorityText[props[3]]);
+    arr.push(Project.mainList[props[4]].title);
+    return arr;
+  }
+
+  populateTaskForm() {
+    const taskInputs = document.querySelectorAll('[data-type="out"]');
+    for (let i = 0; i < taskInputs.length; i += 1) {
+      const element = taskInputs[i];
+      element.value = this.findProps()[i];
+    }
+    return taskInputs;
+  }
+
+  addEditEvent(button) {
+    button.addEventListener('click', () => {
+      this.populateTaskForm();
+      const editTaskForm = document.getElementById('editTaskForm');
+      utils.addHiddenInput(editTaskForm, this);
+      utils.toggleShowElement(editTaskForm);
```

Comment:
- DOM manipulation logic should be separated from pure JS. Move any DOM logic to different file.